dialogue-core/src/main/java/com/palantir/dialogue/core/ConcurrencyLimitedChannel.java

```diff
@@ -76,15 +73,25 @@ static LimitedChannel create(Config cf, LimitedChannel channel, int uriIndex) {
         this.limiter = limiter;
         Preconditions.checkArgument(
                 uriIndex != -1, "uriIndex must be specified", SafeArg.of("channel-name", channelName));
-        weakGauge(
+        DialogueConcurrencylimiterMetrics metrics = DialogueConcurrencylimiterMetrics.of(taggedMetrics);
+        DialogueInternalWeakReducingGauge.getOrCreateDouble(
                 taggedMetrics,
-                DialogueConcurrencylimiterMetrics.of(taggedMetrics)
-                        .max()
+                metrics.max()
                         .channelName(channelName)
                         .hostIndex(Integer.toString(uriIndex))
                         .buildMetricName(),
-                this,
-                ConcurrencyLimitedChannel::getMax);
+                ConcurrencyLimitedChannel::getMax,
+                doubleStream -> doubleStream.max().orElse(0D),
+                this);
+        DialogueInternalWeakReducingGauge.getOrCreate(
```

Comment:
Would it be crazy to wrap this in a `log.isDebugEnabled()` flag? I know it would only enable at startup, just that every metric costs $ and we previously deleted some similar metrics (I think we had active requests, active connections and concurrency limiter inflight, which all pretty much showed the same number)

Reply:
I'd rather have data available for incidents that have already occurred. Days of engineer time are expensive.